### PR TITLE
Add a test to verify that we generate valid .ts code

### PR DESCRIPTION
**Pull request by bowenni**
Last piece of https://github.com/angular/clutz/issues/404
Depends on https://github.com/angular/clutz/pull/476 and https://github.com/angular/clutz/pull/486

### CODE REVIEW

File: src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java [hunk 1]
```diff
@@ -45,9 +45,18 @@ public boolean accept(File dir, String name) {
       new FilenameFilter() {
         @Override
         public boolean accept(File dir, String name) {
-          return name.endsWith(".ts");
+          return name.endsWith(".ts") && !name.endsWith(".d.ts");
```
Comment (gmoothart) [inline src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java]: (optional) could you do something like `&& !D_TS.accept(dir, name)` ?

Comment (bowenni) [inline src/test/java/com/google/javascript/clutz/DeclarationGeneratorTests.java]: Done.

File: src/test/java/com/google/javascript/gents/GeneratedSyntaxTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+public class GeneratedSyntaxTest {
+  /*
+   * Ideally we want to generate both idiomatic and valid TypeScript code. But in some cases we
+   * choose to emit idiomatic TS and sacrifice the correctness.
+   */
+  private static final String EXCLUDED_TS_ARRAY[] = {
+    "classes.ts", "module_namespace.ts", "static_methods.ts", "proto_methods.ts"
+  };
+  private static final Set<String> EXCLUDED_TS =
+      new HashSet<String>(Arrays.asList(EXCLUDED_TS_ARRAY));
```
Comment (rkirov) [inline src/test/java/com/google/javascript/gents/GeneratedSyntaxTest.java]: prefer ImmutableSet.of

File: src/test/java/com/google/javascript/gents/GeneratedSyntaxTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      tscCommand.add(tsInput.getPath());
+      File dtsInput = getDTS(tsInput, dtsInputs);
+      if (dtsInput != null) {
+        tscCommand.add(dtsInput.getPath());
+      }
+      DeclarationSyntaxTest.runChecked(tscCommand);
+    }
+  }
+
+  private static File getDTS(File tsInput, List<File> dtsInputs) {
```
Comment (rkirov) [inline src/test/java/com/google/javascript/gents/GeneratedSyntaxTest.java]: getDts is the preferred capitalization.

Also, mabye name getHelperDts, getAdditionalDts to make it clear these files are optional and just add some types needed for compilation (I was wonder for a bit what the purpose here).

File: src/test/java/com/google/javascript/gents/GeneratedSyntaxTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  private static final FilenameFilter COMPILABLE_TS_SOURCES =
+      new FilenameFilter() {
+        @Override
+        public boolean accept(File dir, String name) {
+          return TS_SOURCES.accept(dir, name) && !EXCLUDED_TS.contains(name);
+        }
+      };
+
+  @Test
+  public void testGenerated() throws Exception {
```
Comment (rkirov) [inline src/test/java/com/google/javascript/gents/GeneratedSyntaxTest.java]: Add a comment (todo?) that this only does singleTests and not multiFileTests

**Pull request merged**